chef/cookbooks/cpe_profiles/resources/cpe_profiles.rb

```diff
@@ -32,6 +32,26 @@
   end
 end
 
+def query_installed_profiles_legacy
+  # 10.7 has a bug where if you pass .plist it makes a second .plist.
+  if node.os_less_than?('10.8')
+    tempfile = ::Dir::Tmpname.create('stdout-xml') {}
+  else
+    tempfile = ::Dir::Tmpname.create('stdout-xml.plist') {}
+  end
+  # Dump all profile metadata to a tempfile
+  shell_out!("profiles -P -o '#{tempfile}'")
+  # 10.7 has a bug where if you pass .plist it makes a second .plist.
+  if node.os_less_than?('10.8')
+    installed_profiles = Plist.parse_xml(tempfile + '.plist')
```

Comment:
You are adding (or not adding) .plist in the tempfile declaration above, so can get away with removing these if statements and relying on 37-41 to get the correct filename (with or without `.plist`). In the case of 10.8+, it might end up as `.plist.plist` otherwise.

This should give you a single line:

`installed_profiles = Plist.parse_xml(tempfile)`

Reply:
10.7 has a bug where if you do `profiles -P -o stdout-xml.plist` you will get a file called stdout-xml.plist.plist. So in order to get around this I pass `profiles -P -o stdout-xml` which then gives me stdout-xml.plist.

However you are reading the code incorrectly. Because I hit the 10.7, my tempfile variable is now missing the `.plist` in the name, so I have to pass it.

Reply:
There is no condition for 10.8+ that could result in `.plist.plist` because the guards that do both have the same check.

Reply:
Oh I see what is going on here. It was a little confusing.

With 37-41, we know it is getting output to the correct filename (`stdout-xml.plist`). I guess it feels repetitive to have another casement to re-append the .plist or not, depending on OS version. 

Is it possible to get the temp directory above made outside the if statement? Maybe:

`tempdir = ::Dir::Tmpname.create`

and then inside the if statement, you'd do `::File.join(tempdir, 'stdout-xml')` or `::File.join(tempdir, 'stdout-xml.plist')` based upon which os version it is.

Then, because we know 10.7 appends .plist and 10.8+ does the right thing, you can read it with one command without the if statement:

`installed_profiles = Plist.parse_xml(::File.join(tempdir, 'stdout-xml.plist'))`  (We should use a File.join here anyways as string math isn't ideal).

Reply:
10.7 is just bad and should feel bad. That is my conclusion here :)